code/contracts/ONEWallet.sol

```diff
@@ -117,35 +118,66 @@ contract ONEWallet is TokenTracker {
+        }
+        bytes32[] memory hashes = new bytes32[](numCommits);
+        bytes32[] memory paramHashes = new bytes32[](numCommits);
+        bytes32[] memory verificationHashes = new bytes32[](numCommits);
+        uint32[] memory timestamps = new uint32[](numCommits);
+        bool[] memory completed = new bool[](numCommits);
+        uint32 index = 0;
+        for (uint32 i = 0; i < commits.length; i++) {
+            Commit[] storage cc = commitLocker[commits[i]];
+            for (uint32 j = 0; j < cc.length; j++) {
+                Commit storage c = cc[j];
+                hashes[index] = c.hash;
+                paramHashes[index] = c.paramsHash;
+                verificationHashes[index] = c.verificationHash;
+                timestamps[index] = c.timestamp;
+                completed[index] = c.completed;
+                index++;
+            }
+        }
+        return (hashes, paramHashes, verificationHashes, timestamps, completed);
+    }
+
+    function findCommit(bytes32 /*hash*/) external pure returns (bytes32, bytes32, uint32, bool){
+        revert("Deprecated");
+    }
+
+    function lookupCommit(bytes32 hash) external view returns (bytes32[] memory, bytes32[] memory, bytes32[] memory, uint32[] memory, bool[] memory){
+        Commit[] storage cc = commitLocker[hash];
+        bytes32[] memory hashes = new bytes32[](cc.length);
+        bytes32[] memory paramHashes = new bytes32[](cc.length);
+        bytes32[] memory verificationHashes = new bytes32[](cc.length);
+        uint32[] memory timestamps = new uint32[](cc.length);
+        bool[] memory completed = new bool[](cc.length);
+        for (uint32 i = 0; i < cc.length; i++) {
+            Commit storage c = cc[i];
             hashes[i] = c.hash;
             paramHashes[i] = c.paramsHash;
+            verificationHashes[i] = c.verificationHash;
             timestamps[i] = c.timestamp;
             completed[i] = c.completed;
         }
-        return (hashes, paramHashes, timestamps, completed);
+        return (hashes, paramHashes, verificationHashes, timestamps, completed);
     }
 
-    function findCommit(bytes32 hash) external view returns (bytes32, bytes32, uint32, bool){
-        Commit storage c = commitLocker[hash];
-        return (c.hash, c.paramsHash, c.timestamp, c.completed);
-    }
-
-    function commit(bytes32 hash, bytes32 paramsHash) external {
+    function commit(bytes32 hash, bytes32 paramsHash, bytes32 verificationHash) external {
         _cleanupCommits();
-        Commit storage c = commitLocker[hash];
-        require(c.timestamp == 0 && !c.completed, "Commit already exists");
-        Commit memory nc = Commit(hash, paramsHash, uint32(block.timestamp), false);
+        Commit memory nc = Commit(hash, paramsHash, verificationHash, uint32(block.timestamp), false);
         require(commits.length < MAX_COMMIT_SIZE, "Too many commits");
         commits.push(hash);
-        commitLocker[hash] = nc;
```

Comment:
The hash under you store to commitLocker is sufficient when it contains only h(EOTP). The true commitment hash that commits to the parameters using EOTP is stored as the verification hash in commit structure. So, removing the neighbor and indexWithNonce from hash should be made. (Also, the hash field can be dropped from the commit structure - as I already mentioned in other review.)

Reply:
Also discussed over chat, and I agree.

Reply:
On a second thought - I think indexWithNonce should be kept. There may be repetitions in EOTP (especially if double OTP and controlled randomness are not used), though the probability that they occur less than two time intervals apart are very unlikely. The computation for 32 bytes more data in keccak256 is cheap anyway. And for backward compatibility reasons, it doesn't hurt to have 32 more bytes for neighbor as well. Let's keep this as is for now.